symbolic-debuginfo/src/ppdb.rs

```diff
@@ -154,8 +172,13 @@ impl<'data> PortablePdbDebugSession<'data> {
     /// Looks up a file's source contents by its full canonicalized path.
     ///
     /// The given path must be canonicalized.
-    pub fn source_by_path(&self, _path: &str) -> Result<Option<Cow<'_, str>>, FormatError> {
-        Ok(None)
+    pub fn source_by_path(&self, path: &str) -> Result<Option<Cow<'_, str>>, FormatError> {
```

Comment:
Hm, given that this is the fn signature of our `DebugSession`, it does make sense to lazily decompress the file contents, and have optional errors there.

I wonder if it might be worth breaking this and making it a `&mut self` fn. That way, the `DebugSession` can maintain an internal cache of already decompressed file contents. It seems a bit suboptimal that we decompress the same file over and over if we have multiple frames that map to the same file.
Though this is a already a problem for our normal sourcebundles.
I wonder if symbolicator itself would even allow this to be `mut`, maybe it wouldn’t like that either.

Reply:
Extracted a standalone issue as this is out of scope IMO and it would benefit source bundles much more.